From: Borislav Petkov <bp@amd64.org>
Subject: x86: Add a BSP cpu_dev helper
References: bnc#711285
Patch-Mainline: queued in x86-tip for 3.2

Acked-by: Mel Gorman <mgorman@novell.com>
Signed-off-by: Thomas Renninger <trenn@suse.de>

Add a function ptr to struct cpu_dev which is destined to be run only
once on the BSP during boot.

Signed-off-by: Borislav Petkov <borislav.petkov@amd.com>
Link: http://lkml.kernel.org/r/20110805180116.GB26217@aftab
Signed-off-by: H. Peter Anvin <hpa@linux.intel.com>

Automatically created from "patches.arch/x86_amd_interlagos_mem_alignment_perf_improve_2_4.patch" by xen-port-patches.py

--- a/arch/x86/kernel/cpu/common-xen.c
+++ b/arch/x86/kernel/cpu/common-xen.c
@@ -731,6 +731,9 @@ static void __init early_identify_cpu(st
 	setup_smep(c);
 
 	fpu__init_parse_early_param();
+
+	if (this_cpu->c_bsp_init)
+		this_cpu->c_bsp_init(c);
 }
 
 void __init early_cpu_init(void)
